units/sdlmutex.inc

```diff
@@ -1,145 +1,238 @@
 // based on "sdl_mutex.h" (2.0.14)
-{ TODO : Add type definitions (see todos below).  }
 
+const
   {**
    *  Synchronization functions which can time out return this value
    *  if they time out.
    *}
-const
   SDL_MUTEX_TIMEDOUT = 1;
 
   {**
    *  This is the timeout value which corresponds to never time out.
    *}
-  //SDL_MUTEX_MAXWAIT   (~(Uint32)0)
+  SDL_MUTEX_MAXWAIT = Not cuint32(0);
 
-
-  {**
-   *  Mutex functions
-   *}
+  { -- Mutex functions -- }
 type
-  {*The SDL mutex structure, defined in SDL_sysmutex.c *}
+  { The SDL mutex structure, defined in SDL_sysmutex.c }
   PPSDL_Mutex = ^PSDL_Mutex;
-  PSDL_Mutex = Pointer; //todo!
+  PSDL_Mutex = Type Pointer;
```

Comment:
This should better be
```
type
  PPSDL_Mutex = ^PSDL_Mutex;
  PSDL_Mutex = ^TSDL_Mutex;
  TSDL_Mutex = record end;
```
right?

Reply:
I see, 
you never stated your opinion on this topic (exact translation of opaque structs), which was discussed in issue #63.

In short: From my point of view, which admittetedly is heavily influenced by a translator-view, the translation of
```
/* The SDL mutex structure, defined in SDL_sysmutex.c */
struct SDL_mutex;
typedef struct SDL_mutex SDL_mutex;
```
is better reflected by
```
TSDL_mutex = record end;
```
rather than by
```
PSDL_mutex = type Pointer;
```

I understand, that formally it would not be necessary to declare the TSDL_mutex type if it is not used in the header code though, but wouldn'd call this necessarily faux'ed. I consider this kind of a dilemma.

What do you think?

Reply:
I didn't think much of it when the question was first asked, basically following a "if it works, don't break it" principle - but over time, I came to favour the `type Pointer` solution, since by avoiding having a `TSDL_Something` declaration in the first place, it's made quite clear the user is not expected to ever use the non-pointer type.

Reply:
I see your point, it would make use of the units safer in some regard. Well, then let's make both ways possible for now. An update of the [code style guidelines](https://github.com/PascalGameDevelopment/SDL2-for-Pascal/blob/master/CHEATSHEET.md) has to be done then. Let me know, if I should do it.

We could think about a goal for a future version of the units to only favor the pointer-translation for consistency.

Best regards